etc/containerpilot.json

```diff
@@ -1,16 +1,53 @@
 {
-  "consul": "{{ .CONSUL }}:8500",
-  "preStart": [
-    "consul-template", "-once", "-consul", "{{ .CONSUL }}:8500", "-template",
-    "/etc/prometheus/prometheus.yml.ctmpl:/etc/prometheus/prometheus.yml"
-  ],
-  "services": [
+  "consul": "localhost:8500",
+  "jobs": [
+    {
+      "name": "setup-config",
+      "exec": [
+        "consul-template", "-once", "-consul", "localhost:8500", "-template",
+        "/etc/prometheus/prometheus.yml.ctmpl:/etc/prometheus/prometheus.yml"
+      ],
+      "when": {
+        "source": "consul-agent",
+        "once": "healthy"
+      }
+    },
     {
       "name": "prometheus",
       "port": 9090,
-      "health": ["curl", "-so", "/dev/null", "http://localhost:9090/metrics"],
-      "poll": 10,
-      "ttl": 25
+      "exec": [
+        "/bin/prometheus",
+        "-config.file=/etc/prometheus/prometheus.yml",
+        "-storage.local.path=/prometheus",
+        "-web.console.libraries=/etc/prometheus/console_libraries",
+        "-web.console.templates=/etc/prometheus/consoles"
+      ],
+      "health": {
+        "exec": "curl -so /dev/null http://localhost:9090/metrics",
+        "interval": 10,
+        "ttl": 25
+      },
+      "when": {
+        "source": "setup-config",
+        "once": "exitSuccess"
+      }
+    },
+    {
+      "name": "consul-agent",
+      "exec": ["/usr/local/bin/consul", "agent",
+                  "-data-dir=/data",
+                  "-config-dir=/config",
+                  "-log-level=err",
+                  "-rejoin",
+                  "-retry-join", "{{ .CONSUL | default "consul" }}",
```

Comment:
This assumes that we always have a Consul agent inside the container, which we need on Triton certainly but not when users reuse this blueprint in non-Triton environments.